var/spack/repos/builtin/packages/srcml/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Srcml(CMakePackage):
+    """srcML is an XML format for source code. The XML markup identifies elements of
+    the abstract syntax of the source-code language. The toolkit includes parsing
+    supports conversion of C, C++, C#, and Java both to and from the srcML format.
+    The format allows leveraging XML tools to support the various tasks of source
+    code exploration, analysis, and manipulation."""
+
+    homepage = "https://github.com/srcML/srcML"
+    url = "https://github.com/srcML/srcML/archive/refs/tags/v1.0.0.tar.gz"
+
+    maintainers = ["meyersbs"]
+
+    version("1.0.0", sha256="3ddf33271c3b3953d5e3ecbb14c4f925fc0e609a81250d921d3516537dcffae2")
+
+    depends_on("cmake@3.14:", type="build")
+    depends_on("antlr+cxx+java+pic")
+    depends_on("libxslt")
+    depends_on("libarchive@3:")
+    depends_on("curl")
+    depends_on("boost")
```

Comment:
This will build for `boost@:1.78.0`, but won't build locally for me with `boost@1.79.0` or `boost@1.80.0`. I get the following errors:

```
==> Installing srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke
==> No binary for srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke found: installing from source
==> Using cached archive: /dev/shm/bsmits_spack/var/spack/cache/_source-cache/archive/3d/3ddf33271c3b3953d5e3ecbb14c4f925fc0e609a81250d921d3516537dcffae2.tar.gz
==> Using cached archive: /dev/shm/bsmits_spack/var/spack/cache/_source-cache/archive/38/384068e00a01809cdc9b6eca79fd6833bf3214d4b9ac1765b52bc374a7af333e
==> Applied patch https://patch-diff.githubusercontent.com/raw/srcML/srcML/pull/1829.patch?full_index=1
==> Ran patch() for srcml
==> srcml: Executing phase: 'cmake'
==> srcml: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

14 errors found in build log:
     192    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/src_output_libarchive.cpp.o -MF CMakeFiles/srcml.dir/src_output_libarchive.cpp.o.d -o CMakeFiles/srcml.dir/src_output_libarchive.cpp.o -c /dev/shm/spack/stage/bsmits/spack-stage
            -srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_output_libarchive.cpp
     193    [ 78%] Building CXX object src/client/CMakeFiles/srcml.dir/src_prefix.cpp.o
     194    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/src_prefix.cpp.o -MF CMakeFiles/srcml.dir/src_prefix.cpp.o.d -o CMakeFiles/srcml.dir/src_prefix.cpp.o -c /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuiha
            vug3ujmlb7ke/spack-src/src/client/src_prefix.cpp
     195    [ 80%] Building CXX object src/client/CMakeFiles/srcml.dir/srcml.cpp.o
     196    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/srcml.cpp.o -MF CMakeFiles/srcml.dir/srcml.cpp.o.d -o CMakeFiles/srcml.dir/srcml.cpp.o -c /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/sp
            ack-src/src/client/srcml.cpp
     197    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp: In function 'int src_input_file(ParseQueue&, srcml_archive*, const srcml_request_t&, const srcml_input_src&)':
  >> 198    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp:41:44: error: no match for 'operator=' (operand types are 'boost::optional<std::__cxx11::basic_string<char> >' and '
            const srcml_output_dest' {aka 'const srcml_input_src'})
     199       41 |         prequest->disk_dir = srcml_request.output_filename;
     200          |                                            ^~~~~~~~~~~~~~~
     201    In file included from /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional.hpp:15,
     202                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./srcml_cli.hpp:29,
     203                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./src_input_file.hpp:27,
     204                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp:23:
     205    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note: candidate: 'template<class Expr> typename boost::enable_if<boost::optional_d
            etail::is_optional_val_assign_candidate<T, Expr>, boost::optional<T>&>::type boost::optional<T>::operator=(Expr&&) [with Expr = Expr; T = std::__cxx11::basic_string<char>]'
     206     1056 |     operator= ( Expr&& expr )
     207          |     ^~~~~~~~
     208    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note:   template argument deduction/substitution failed:
     209    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class Expr> typename boost::enable_if<boost::optional_detail
            ::is_optional_val_assign_candidate<std::__cxx11::basic_string<char>, Expr, boost::optional_detail::has_dedicated_constructor<std::__cxx11::basic_string<char>, Expr>::value>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::o
            ptional<std::__cxx11::basic_string<char> >::operator=<Expr>(Expr&&) [with Expr = const srcml_input_src&]':
     210    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp:41:44:   required from here
  >> 211    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: error: no type named 'type' in 'struct boost::enable_if<boost::optional_detail::is
            _optional_val_assign_candidate<std::__cxx11::basic_string<char>, const srcml_input_src&, false>, boost::optional<std::__cxx11::basic_string<char> >&>'
     212    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note: candidate: 'template<class U> boost::optional<T>& boost::optional<T>::opera
            tor=(const boost::optional<U>&) [with U = U; T = std::__cxx11::basic_string<char>]'
     213     1076 |     optional& operator= ( optional<U> const& rhs )
     214          |               ^~~~~~~~
     215    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note:   template argument deduction/substitution failed:
     216    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp:41:44: note:   'const srcml_output_dest' {aka 'const srcml_input_src'} is not derived from 'const boost::optional<T>
            '
     217       41 |         prequest->disk_dir = srcml_request.output_filename;
     ...

     246    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note: candidate: 'template<class T_> typename boost::enable_if<boost::is_same<T, t
            ypename boost::decay<T_>::type>, boost::optional<T>&>::type boost::optional<T>::operator=(T_&&) [with T_ = T_; T = std::__cxx11::basic_string<char>]'
     247     1127 |     operator= ( T_&& val )
     248          |     ^~~~~~~~
     249    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note:   template argument deduction/substitution failed:
     250    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class T_> typename boost::enable_if<boost::is_same<std::__cx
            x11::basic_string<char>, typename boost::decay<T>::type>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::optional<std::__cxx11::basic_string<char> >::operator=<T_>(T_&&) [with T_ = const srcml_input_src&]':
     251    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_file.cpp:41:44:   required from here
  >> 252    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: error: no type named 'type' in 'struct boost::enable_if<boost::is_same<std::__cxx1
            1::basic_string<char>, srcml_input_src>, boost::optional<std::__cxx11::basic_string<char> >&>'
     253    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:15: note: candidate: 'boost::optional<T>& boost::optional<T>::operator=(boost::none_t
            ) [with T = std::__cxx11::basic_string<char>]'
     254     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     255          |               ^~~~~~~~
     256    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:34: note:   no known conversion for argument 1 from 'const srcml_output_dest' {aka 'c
            onst srcml_input_src'} to 'boost::none_t'
     257     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     258          |                           ~~~~~~~^~~~~
     259    [ 81%] Building CXX object src/client/CMakeFiles/srcml.dir/srcml_cli.cpp.o
     260    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/srcml_cli.cpp.o -MF CMakeFiles/srcml.dir/srcml_cli.cpp.o.d -o CMakeFiles/srcml.dir/srcml_cli.cpp.o -c /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug
            3ujmlb7ke/spack-src/src/client/srcml_cli.cpp
     261    [ 83%] Building CXX object src/client/CMakeFiles/srcml.dir/srcml_consume.cpp.o
     262    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/srcml_consume.cpp.o -MF CMakeFiles/srcml.dir/srcml_consume.cpp.o.d -o CMakeFiles/srcml.dir/srcml_consume.cpp.o -c /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3ie
            pthoiuihavug3ujmlb7ke/spack-src/src/client/srcml_consume.cpp
     263    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp: In function 'int src_input_libarchive(ParseQueue&, srcml_archive*, const srcml_request_t&, const srcml_input_
            src&)':
  >> 264    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp:231:48: error: no match for 'operator=' (operand types are 'boost::optional<std::__cxx11::basic_string<char> >
            ' and 'const srcml_output_dest' {aka 'const srcml_input_src'})
     265      231 |             prequest->disk_dir = srcml_request.output_filename;
     266          |                                                ^~~~~~~~~~~~~~~
     267    In file included from /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional.hpp:15,
     268                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./srcml_cli.hpp:29,
     269                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./src_input_libarchive.hpp:29,
     270                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp:28:
     271    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note: candidate: 'template<class Expr> typename boost::enable_if<boost::optional_d
            etail::is_optional_val_assign_candidate<T, Expr>, boost::optional<T>&>::type boost::optional<T>::operator=(Expr&&) [with Expr = Expr; T = std::__cxx11::basic_string<char>]'
     272     1056 |     operator= ( Expr&& expr )
     273          |     ^~~~~~~~
     274    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note:   template argument deduction/substitution failed:
     275    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class Expr> typename boost::enable_if<boost::optional_detail
            ::is_optional_val_assign_candidate<std::__cxx11::basic_string<char>, Expr, boost::optional_detail::has_dedicated_constructor<std::__cxx11::basic_string<char>, Expr>::value>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::o
            ptional<std::__cxx11::basic_string<char> >::operator=<Expr>(Expr&&) [with Expr = const srcml_input_src&]':
     276    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp:231:48:   required from here
  >> 277    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: error: no type named 'type' in 'struct boost::enable_if<boost::optional_detail::is
            _optional_val_assign_candidate<std::__cxx11::basic_string<char>, const srcml_input_src&, false>, boost::optional<std::__cxx11::basic_string<char> >&>'
     278    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note: candidate: 'template<class U> boost::optional<T>& boost::optional<T>::opera
            tor=(const boost::optional<U>&) [with U = U; T = std::__cxx11::basic_string<char>]'
     279     1076 |     optional& operator= ( optional<U> const& rhs )
     280          |               ^~~~~~~~
     281    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note:   template argument deduction/substitution failed:
     282    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp:231:48: note:   'const srcml_output_dest' {aka 'const srcml_input_src'} is not derived from 'const boost::opti
            onal<T>'
     283      231 |             prequest->disk_dir = srcml_request.output_filename;

     ...

     316    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note: candidate: 'template<class T_> typename boost::enable_if<boost::is_same<T, t
            ypename boost::decay<T_>::type>, boost::optional<T>&>::type boost::optional<T>::operator=(T_&&) [with T_ = T_; T = std::__cxx11::basic_string<char>]'
     317     1127 |     operator= ( T_&& val )
     318          |     ^~~~~~~~
     319    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note:   template argument deduction/substitution failed:
     320    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class T_> typename boost::enable_if<boost::is_same<std::__cx
            x11::basic_string<char>, typename boost::decay<T>::type>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::optional<std::__cxx11::basic_string<char> >::operator=<T_>(T_&&) [with T_ = const srcml_input_src&]':
     321    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_libarchive.cpp:231:48:   required from here
  >> 322    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: error: no type named 'type' in 'struct boost::enable_if<boost::is_same<std::__cxx1
            1::basic_string<char>, srcml_input_src>, boost::optional<std::__cxx11::basic_string<char> >&>'
     323    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:15: note: candidate: 'boost::optional<T>& boost::optional<T>::operator=(boost::none_t
            ) [with T = std::__cxx11::basic_string<char>]'
     324     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     325          |               ^~~~~~~~
     326    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:34: note:   no known conversion for argument 1 from 'const srcml_output_dest' {aka 'c
            onst srcml_input_src'} to 'boost::none_t'
     327     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     328          |                           ~~~~~~~^~~~~
     329    [ 84%] Building CXX object src/client/CMakeFiles/srcml.dir/srcml_display_metadata.cpp.o
     330    cd /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/src/client && /dev/shm/bsmits_spack/lib/spack/env/gcc/g++ -DWITH_LIBXSLT -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyu
            vl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/. -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv/external -I/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihav
            ug3ujmlb7ke/spack-src/src/libsrcml -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libarchive-3.5.2-gnpb42yvqzkx5uxccgjsczldpqzpxtld/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libxml
            2-2.10.3-w4atzweinoskxwmopexjsg7ymb4p4hp5/include/libxml2 -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/libiconv-1.16-mrnnkstgsm7kp2x764wrlbx7yxtbrklz/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx
            512/gcc-9.3.0/curl-7.85.0-5niddbul5ju2oujaidpl3r3eifdk33wc/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include -I/dev/shm/bsmits_spack/opt/spack/linux-rhel7-sky
            lake_avx512/gcc-9.3.0/libxslt-1.1.33-5vswcvlhvkmt5psgpmdhq46ib42lba7f/include -O2 -g -DNDEBUG -fPIC -Wno-long-long -Wall -Wextra -pedantic -Wempty-body -Wignored-qualifiers -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pragmas -Wno-varia
            dic-macros -std=gnu++11 -MD -MT src/client/CMakeFiles/srcml.dir/srcml_display_metadata.cpp.o -MF CMakeFiles/srcml.dir/srcml_display_metadata.cpp.o.d -o CMakeFiles/srcml.dir/srcml_display_metadata.cpp.o -c /dev/shm/spack/stage/bsmits/spack-st
            age-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/srcml_display_metadata.cpp
  >> 331    make[2]: *** [src/client/CMakeFiles/srcml.dir/build.make:275: src/client/CMakeFiles/srcml.dir/src_input_file.cpp.o] Error 1
     332    make[2]: *** Waiting for unfinished jobs....
  >> 333    make[2]: *** [src/client/CMakeFiles/srcml.dir/build.make:317: src/client/CMakeFiles/srcml.dir/src_input_libarchive.cpp.o] Error 1
     334    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp: In function 'int src_input_text(ParseQueue&, srcml_archive*, const srcml_request_t&, const srcml_input_src&)':
  >> 335    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp:70:48: error: no match for 'operator=' (operand types are 'boost::optional<std::__cxx11::basic_string<char> >' and '
            const srcml_output_dest' {aka 'const srcml_input_src'})
     336       70 |             prequest->disk_dir = srcml_request.output_filename;
     337          |                                                ^~~~~~~~~~~~~~~
     338    In file included from /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional.hpp:15,
     339                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./srcml_cli.hpp:29,
     340                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/./src_input_text.hpp:27,
     341                     from /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp:23:
     342    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note: candidate: 'template<class Expr> typename boost::enable_if<boost::optional_d
            etail::is_optional_val_assign_candidate<T, Expr>, boost::optional<T>&>::type boost::optional<T>::operator=(Expr&&) [with Expr = Expr; T = std::__cxx11::basic_string<char>]'
     343     1056 |     operator= ( Expr&& expr )
     344          |     ^~~~~~~~
     345    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: note:   template argument deduction/substitution failed:
     346    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class Expr> typename boost::enable_if<boost::optional_detail
            ::is_optional_val_assign_candidate<std::__cxx11::basic_string<char>, Expr, boost::optional_detail::has_dedicated_constructor<std::__cxx11::basic_string<char>, Expr>::value>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::o
            ptional<std::__cxx11::basic_string<char> >::operator=<Expr>(Expr&&) [with Expr = const srcml_input_src&]':
     347    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp:70:48:   required from here
  >> 348    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1056:5: error: no type named 'type' in 'struct boost::enable_if<boost::optional_detail::is
            _optional_val_assign_candidate<std::__cxx11::basic_string<char>, const srcml_input_src&, false>, boost::optional<std::__cxx11::basic_string<char> >&>'
     349    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note: candidate: 'template<class U> boost::optional<T>& boost::optional<T>::opera
            tor=(const boost::optional<U>&) [with U = U; T = std::__cxx11::basic_string<char>]'
     350     1076 |     optional& operator= ( optional<U> const& rhs )
     351          |               ^~~~~~~~
     352    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1076:15: note:   template argument deduction/substitution failed:
     353    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp:70:48: note:   'const srcml_output_dest' {aka 'const srcml_input_src'} is not derived from 'const boost::optional<T>
            '
     354       70 |             prequest->disk_dir = srcml_request.output_filename;

     ...
     ...

     383    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note: candidate: 'template<class T_> typename boost::enable_if<boost::is_same<T, t
            ypename boost::decay<T_>::type>, boost::optional<T>&>::type boost::optional<T>::operator=(T_&&) [with T_ = T_; T = std::__cxx11::basic_string<char>]'
     384     1127 |     operator= ( T_&& val )
     385          |     ^~~~~~~~
     386    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: note:   template argument deduction/substitution failed:
     387    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp: In substitution of 'template<class T_> typename boost::enable_if<boost::is_same<std::__cx
            x11::basic_string<char>, typename boost::decay<T>::type>, boost::optional<std::__cxx11::basic_string<char> >&>::type boost::optional<std::__cxx11::basic_string<char> >::operator=<T_>(T_&&) [with T_ = const srcml_input_src&]':
     388    /dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-src/src/client/src_input_text.cpp:70:48:   required from here
  >> 389    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1127:5: error: no type named 'type' in 'struct boost::enable_if<boost::is_same<std::__cxx1
            1::basic_string<char>, srcml_input_src>, boost::optional<std::__cxx11::basic_string<char> >&>'
     390    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:15: note: candidate: 'boost::optional<T>& boost::optional<T>::operator=(boost::none_t
            ) [with T = std::__cxx11::basic_string<char>]'
     391     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     392          |               ^~~~~~~~
     393    /dev/shm/bsmits_spack/opt/spack/linux-rhel7-skylake_avx512/gcc-9.3.0/boost-1.79.0-zogutfpogci7po7z2bgvzl6k2atmb265/include/boost/optional/optional.hpp:1157:34: note:   no known conversion for argument 1 from 'const srcml_output_dest' {aka 'c
            onst srcml_input_src'} to 'boost::none_t'
     394     1157 |     optional& operator= ( none_t none_ ) BOOST_NOEXCEPT
     395          |                           ~~~~~~~^~~~~
  >> 396    make[2]: *** [src/client/CMakeFiles/srcml.dir/build.make:331: src/client/CMakeFiles/srcml.dir/src_input_text.cpp.o] Error 1
     397    make[2]: Leaving directory '/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv'
  >> 398    make[1]: *** [CMakeFiles/Makefile2:376: src/client/CMakeFiles/srcml.dir/all] Error 2
     399    make[1]: Leaving directory '/dev/shm/spack/stage/bsmits/spack-stage-srcml-1.0.0-vvulyuvl3iepthoiuihavug3ujmlb7ke/spack-build-vvulyuv'
  >> 400    make: *** [Makefile:149: all] Error 2
```